SysArchMeter.c

```diff
@@ -9,67 +9,75 @@ in the source distribution for its full text.
 #include "SysArchMeter.h"
 
 #include <stdio.h>
-#include <string.h>
 #include <sys/utsname.h>
 
 #include "XUtils.h"
 
 
 static const int SysArchMeter_attributes[] = {HOSTNAME};
 
-static void SysArchMeter_updateValues(Meter* this, char* buffer, size_t size) {
-   static struct utsname uname_info;
-   static int uname_result;
-   static char distro[3][64] = { {'\0'}, {'\0'}, {'\0'} };
-   static bool loaded_data = false;
+static void parseOSRelease(char* buffer, size_t bufferLen) {
+   FILE* stream = fopen("/etc/os-release", "r");
+   if (!stream) {
+      stream = fopen("/usr/lib/os-release", "r");
+      if (!stream) {
+         xSnprintf(buffer, bufferLen, "Unknown Distro");
+         return;
+      }
+   }
 
-   (void)this;
+   char name[64] = {'\0'};
+   char version[64] = {'\0'};
+   char lineBuffer[256];
+   while (fgets(lineBuffer, sizeof(lineBuffer), stream)) {
+      if (String_startsWith(lineBuffer, "PRETTY_NAME=\"")) {
+         size_t start = strlen("PRETTY_NAME=\"");
+         size_t len = strrchr(lineBuffer, '"') - lineBuffer - start;
```

Comment:
This seems off when no second double quote character is in that string.

Reply:
Since we matched `PRETTY_NAME=\"`, which contains a `"`, `strrchr()` can not return `NULL` and len might be at least `0`.

Reply:
In which case it still would return `len=-1` …